### ISSUE TITLE
Error when trying to read a prod3 file

### ISSUE THREAD

**Issue by watsonjj** [cta-observatory/pyeventio#137]
I have an old prod3 file which I could read with pyhessio, but an error is raised with pyeventio:

```python
In [1]: from ctapipe.io import SimTelEventSource
   ...: r = SimTelEventSource(input_url="/Users/Jason/Data/sim_telarray/gamma_20deg_0deg_run10251___cta-prod3-merged_desert-2150m-Paranal-subarray-3_cone10.simtel.gz")
   ...: for e in r:
   ...:     pass
   ...:
---------------------------------------------------------------------------
WrongType                                 Traceback (most recent call last)
<ipython-input-1-1e9fcff7bae9> in <module>()
      1 from ctapipe.io import SimTelEventSource
----> 2 r = SimTelEventSource(input_url="/Users/Jason/Data/sim_telarray/gamma_20deg_0deg_run10251___cta-prod3-merged_desert-2150m-Paranal-subarray-3_cone10.simtel.gz")
      3 for e in r:
      4     pass

~/Software/ctapipe/ctapipe/io/simteleventsource.py in __init__(self, config, tool, **kwargs)
     20         super().__init__(config=config, tool=tool, **kwargs)
     21         self.metadata['is_simulation'] = True
---> 22         self.file_ = SimTelFile(self.input_url)
     23
     24         self._subarray_info = self.prepare_subarray_info(

~/anaconda3/envs/cta/lib/python3.6/site-packages/eventio/simtel/simtelfile.py in __init__(self, path, allowed_telescopes)
     71             o = next(self)
     72
---> 73         check_type(o, RunHeader)
     74         self.header = o.parse()
     75         self.n_telescopes = self.header['n_telescopes']

~/anaconda3/envs/cta/lib/python3.6/site-packages/eventio/exceptions.py in check_type(obj, expected)
     36 def check_type(obj, expected):
     37     if not isinstance(obj, expected):
---> 38         raise WrongType(expected, obj)

WrongType: Expected Object RunHeader with type 2000 but received MCRunHeader with type 2001
```

The file is 1GB, but I have uploaded it here: 
https://mcloud.mpi-hd.mpg.de/index.php/s/9C8j46i83jx7c9j/download

**Comment by watsonjj** [comment]
I am using pyeventio 0.14.0

**Comment by dneise** [comment]
Thanks for reporting .. looking into it.

**Comment by dneise** [comment]
Hello the structure of the object seems really strange to me, but still I think the branch in PR #139 could be used to play and see if it works.

Tests are still missing.

However I must say, I am really puzzled by the way the objects have been stored in that file. Looks like somebody just took the file and shook it until some objects ended up in strange places....

Clearly there is something I have to learn still about the simtel file format

**Comment by watsonjj** [comment]
I'm afraid I don't know the details of how this file was created.

**Comment by vuillaut** [comment]
Not much info on how they were produced but that's all I got:
https://forge.in2p3.fr/projects/cta_dirac/wiki/CTA-DIRAC_MC_PROD3_Status

**Comment by dneise** [comment]
My first shot at the implementation of `eventio/simtel/simtelfile.py` was very strict about the order of objects .... then @MaxNoe looked at it and removed many of the requirements I had put in ... also making it more readable.

I start to assume we must require even less structure in the order of the objects within the file.

You know .. one has a mental model of a file format like ... first there is a run header ... then an event header .. then an event .. another event header .. another event .. and so on.

It would be difficult for us to allow events to occur *before* a run header for example. However I believe we can still get rid of some more of these requirements. Of course this weakens our handle on decideing whether a file is broken or not.

I will open another PR next to #139 where I try to get rid of as many requirements as possible.

**Comment by dneise** [comment]
In the meantime maybe you'd still like to explore a bit #139.

**Comment by dneise** [comment]
> I'm afraid I don't know the details of how this file was created.

>Not much info on how they were produced but that's all I got:
https://forge.in2p3.fr/projects/cta_dirac/wiki/CTA-DIRAC_MC_PROD3_Status

So ... the file name contains the string "merged" ... could that be a hint, that this file has been modified after its original creation and during this modification .. something strange happened? 
However the original pyhessio reader does not care at all in which order objects appear in the file, so nobody ever realized?

**Comment by MaxNoe** [comment]
Should be fixed in v0.16.1
